aten/src/ATen/native/cpu/FlashAttentionKernel.cpp

```diff
@@ -21,6 +21,62 @@ namespace at::native {
 
 namespace {
 
+// out = val * a + b
+template <typename T1, typename T2,
+          typename std::enable_if_t<std::is_same_v<T1, float> && is_reduced_floating_point<T2>::value, int> = 0>
+inline void _scale_attn_mask_fusion_kernel(
+    T1* a,
+    T2* b,
+    const int& size,
+    T1* out,
+    T1& val) {
+  auto vec_size1 = at::vec::Vectorized<T1>::size();
+  auto vec_size2 = at::vec::Vectorized<T2>::size();
+  auto vec_scale = at::vec::Vectorized<T1>(val);
+  int64_t i = 0;
+  for (; i < size - (size % vec_size2); i += vec_size2) {
+    auto a0 = at::vec::Vectorized<T1>::loadu(a + i);
+    auto a1 = at::vec::Vectorized<T1>::loadu(a + i + vec_size1);
+    auto b_tmp = at::vec::Vectorized<T2>::loadu(b + i);
+    auto [b0, b1] = convert_to_float<T2>(b_tmp);
```

Comment:
You can probably unify the two functions into one the vec convert too, with vectorizedn.

Reply:
Done. Please take a look.